### DISCUSSION TITLE
Sveltekit getUserByCookie()

### DISCUSSION THREAD

**Discussion by JVariance** [supabase/supabase#1813] [Questions] [answered]
In my handle function in hooks.js I use the result of `cookie.parse(request.headers.cookie)` as argument for supabase.auth.api.getUserByCookie()
But Supabase says that the cookie can't be parsed and if I use express I shall use the middleware 'cookie-parser'.
The parsing of the cookie works well though.
Or can't I use this function with sveltekit?

**Comment by ashleyconnor** [comment, marked as answer]
Try something like:

```javascript
export async function getSession(request) {
  const cookies = cookie.parse(request.headers.cookie || "");

  const { user } = await supabase.auth.api.getUserByCookie({
    ...request,
    cookies
  });

  return {
    user,
  };
}
```

SvelteKit doesn't use Express style request/response objects.

**Reply by JVariance** [reply]
Thanks, I'll try this 🙌🏻

**Reply by ashleyconnor** [reply]
I was working on this last night and I've put together a more comprehensive answer here: https://github.com/supabase/supabase/discussions/1977#discussioncomment-906692
